docs/topics/pages.md

```diff
@@ -493,3 +493,18 @@ class EventPage(Page):
 
     objects = EventPageManager()
 ```
+
+### Support for proxy page models
+
+```{versionadded} 5.0```
+
+[Proxy models](https://docs.djangoproject.com/en/stable/topics/db/models/#proxy-models) are a way to model polymorphism in Django; where multiple variations of a model share the same fields (defined on a concrete 'parent' model) but differ in other ways. For example, they might implement one or more model methods differently, or set different values on the class to affect certain behaviours (e.g. `template_name`, `search_fields`, `content_panels`).
+
+Because Wagtail stores a reference to the `ContentType` for every page it creates, then uses it to recall pages the page as an instance of that type, proxy models work in exactly the same way as concrete ones. Wagtail simply stores the `ContentType` for the proxy model, and returns an instance of that model when requested.
+
+Unless you are using the `Page` model as the concrete parent, at least one 'join' will be required at the database level to retrieve your pages, so the usual benefit of proxy model use is somewhat diminished. However, proxy page models come with some additional benefits:
+
+1. Using `Page.specific` on an instance of the concrete parent model will assemble the proxy model instance entirely in Python code, without querying the database.
+2. Where proxy model instances are detected in a queryset `PageQuerySet.specific()` will reduce the number of queries it makes to fetch specific pages, instead preferring to 'upgrade' proxy model instances in Python as the queryset is evaluated.
+
+To top it all off, the `PageManager` will automatically detect when a queryset has been requested from a proxy model, and filter it to only include pages of that type. The (often unexpected) default behaviour for proxy model managers is to return **all entries** from the parent model table, regardless of which model was used to save the data originally. If you are thinking of registering proxy models as Snippets or with `wagtail.contrib.modeladmin`, this is something you'll need to be aware of.
```

Comment:
@zerolab [suggested](https://github.com/wagtail/wagtail/pull/9255#pullrequestreview-1143310480) documentation

> to capture the need for using for_concrete_models=False whenever we are using ContentType queries as that will likely be something we will overlook with time. Perhaps one for dev docs?

Unless I missed where that was added elsewhere, maybe that makes sense to add at the end here? Something like:

> If you write code that queries against ContentType models in some way, you'll want to be sure to pass `for_concrete_models=False` ([for example](https://docs.djangoproject.com/en/4.1/ref/contrib/contenttypes/#django.contrib.contenttypes.models.ContentTypeManager.get_for_model)) to ensure that you are retrieving proxy model instances.

Reply:
To be honest, I'm not really sure that's good advice, as it's really more nuanced than that.

For permission-related code, you should probably always use `for_concrete_model=False`. 

For something like pages, where you want to know the exact ContentType for each object, you probably also want to use `for_concrete_model=False`.

However, there are situations where it's better to store/use the concrete `ContentType` only. For example, when storing translations for instances of proxy models, where the translations are relevant to ALL variations of that instance, rather than just one.

I don't think we can realistically get away from the fact that this is something that needs to be considered on a case-by-case basis. I would suggest we just rely on the test suite to tell us about breakages, and try to keep it in mind during review.

Reply:
Of course, you're right. Thanks for the explanation.

Reply:
Thank you for following up on this both.

I can see myself looking back at the

> For permission-related code, you should probably always use for_concrete_model=False.
>
> For something like pages, where you want to know the exact ContentType for each object, you probably also want to use for_concrete_model=False.
>
> However, there are situations where it's better to store/use the concrete ContentType only. For example, when storing/finding translations for instances of proxy models, where the translations are relevant to ALL variations of that instance, rather than just one.

reference as there's only so much context I can keep in my head.

I agree it is in the grey area and there isn't a good place to cover all bases and there isn't a hard and fast rule either.

Happy with "what about proxy models?" nagging during PRs